codalab/lib/worksheet_util.py

```diff
@@ -167,24 +167,33 @@ def request_lines(worksheet_info, client):
         raise UsageError('No change made; aborting')
     return form_result
 
+def get_bundle_uuids(client, worksheet_uuid, bundle_specs):
+    '''
+    Return the bundle_uuids corresponding to bundle_specs.
+    Important difference from client.get_bundle_uuids: if a bundle_spec is already
+    a uuid, then just return it directly.  This avoids an extra call to the
+    client.
+    '''
+    resolved = []
+    unresolved = []
+    for bundle_spec in bundle_specs:
+        bundle_spec = bundle_spec.strip()
+
+        if spec_util.UUID_REGEX.match(bundle_spec):
+            resolved.append(bundle_spec)
+        else:
+            unresolved.append(bundle_spec)
+
+    return resolved + client.get_bundle_uuids(worksheet_uuid, unresolved)
```

Comment:
Don't we need to return the uuids in the same order as the bundle_specs?
This code seems might permute things.